palindrom-client.html

```diff
@@ -284,67 +290,84 @@
+                                        removed: (result && [result.removed]) || [],
+                                        addedCount: 0,
+                                        object: parent,
+                                        type: 'splice'
+                                    });
                                     break;
                             }
                         } else {
                             switch (operation.op) {
-
                                 case 'add':
                                 case 'replace':
                                     if (typeof operation.value === 'object') {
-                                        templateDomBind.notifyPath(polymerPath);
+                                        notifications[polymerPath] = [];
                                     } else {
-                                        templateDomBind.notifyPath(polymerPath, operation.value, false);
+                                        notifications[polymerPath] = [operation.value, false];
                                     }
                                     break;
                                 case 'remove':
-                                    //console.warn('Polymer does not support unsetting properties https://github.com/Polymer/polymer/issues/2565');
-                                    //console.warn('remove is translated to JSON incompatible set to `undefined`, as there is no remove method in Polymer.')
-                                    templateDomBind.notifyPath(polymerPath, null, false);
-                                    templateDomBind.notifyPath(polymerPath, undefined, false);
+                                    notifications[polymerPath] = [null, false];
                                     break;
                             }
                         }
                     }
                 }
+                /**
+                *  STEP 2: APPLY THE UPDATES
+                * */
+                const spliceEntries = Object.entries(splices);
+                const notificationEntries = Object.entries(notifications);
+                if (spliceEntries.length + notificationEntries.length > 100) {
+                    console.warn(`palindrom-client has noticed a huge patch of ${patchesAndResults.patches.length} operations, for better performance palindrom-client will replace the whole palindrom object instead of applying your updates individually`);
+                    // clone to prevent flash of content
+                    templateDomBind.set('model', JSON.parse(JSON.stringify(this.obj)));
+                    // Polymer 2 is sync
+                    if (Polymer.Element) {
+                        templateDomBind.set('model', this.obj);
+                    } else { // Polymer 1 is async
+                        setTimeout(() => templateDomBind.set('model', this.obj));
+                    }
+                } else {
```

Comment:
Basically, if there are more unique paths than 100, the whole iteration over the operations was unnecessary because it is the notifications are not made.

I thought it would be simple enough to check if there are more than 100 operations in the patch and then reset the bindings before even attempting to iterate.

Reply:
You're right. I gave up on this too early. I tried resetting the object all the time and you saw how the app blew-up. I think adding an `if N > 100` before doing that would mitigate the problem.